### ISSUE TITLE
Test use of shared libraries / Penpot

### ISSUE THREAD

**Issue by GBKS** [GBKS/bitcoin-wallet-ui-kit#46]
This should be easier to manage overall.

**Comment by editwentyone** [comment]
would love to this

**Comment by editwentyone** [comment]
https://penpot.app/libraries-templates

https://help.penpot.app/user-guide/libraries/#shared-libraries

**Comment by stackingsaunter** [comment]
>Is it possible to push updates to a main component in a separate shared library (and possibly break them in all other files that rely on this library)?

Yes. There's a dialog to confirm the action.
![CleanShot 2023-08-20 at 03 05 58@2x](https://github.com/GBKS/bitcoin-wallet-ui-kit/assets/68239231/7585d9ba-36b6-4e18-9381-985613e681e7)

In a separate file, I updated the component and it changed immediately in the shared library. However, inly the icon changed, and the color didn't.

![CleanShot 2023-08-20 at 03 11 28@2x](https://github.com/GBKS/bitcoin-wallet-ui-kit/assets/68239231/a9e740e9-664b-45dd-b97a-0508ba0b59c0)

![CleanShot 2023-08-20 at 03 10 57@2x](https://github.com/GBKS/bitcoin-wallet-ui-kit/assets/68239231/e69136d2-b626-4b4e-92ce-0f0da0ad8227)

I did similar with Small icon text button, this time color changed, but the copy didn't:
![CleanShot 2023-08-20 at 03 15 13@2x](https://github.com/GBKS/bitcoin-wallet-ui-kit/assets/68239231/54756977-76ec-44a1-9f60-8021d6c3a978)
![CleanShot 2023-08-20 at 03 17 31@2x](https://github.com/GBKS/bitcoin-wallet-ui-kit/assets/68239231/5611dc25-9760-47bd-a663-106cb1af9c20)

**Comment by stackingsaunter** [comment]
>If the UI kit essentially consists of three files, how do I duplicate the set and have the references be updated (so my own new file with screens draws from my own component and icons files)?

I duplicated whole test team to my local "Default" team. Copied files are already a shared library and there's no reference to the original files.

**Comment by stackingsaunter** [comment]
>What is the performance impact? Good, bad, same?

I found performance way better, however that may be affected by other optimisations I did with my browser (I had an extension which slowed down some sites), so I am not sure how it was it. Jumping now to our original files, it seems that it was mostly the extension fault, but performance still feels better in the test files.

**Comment by stackingsaunter** [comment]
>We found that the asset search is not 100% thorough. Does this affect the use of shared libraries?

Yes, works the same, for me it's useless. But perhaps it can be fixed If we work out componenets naming better.

Eg, typing buttons shows only those:
![CleanShot 2023-08-20 at 03 34 47@2x](https://github.com/GBKS/bitcoin-wallet-ui-kit/assets/68239231/54028cd9-ac67-4707-aa3a-2e03b0b47f4b)

**Comment by stackingsaunter** [comment]
>Do they work across teams? If I am part of multiple teams, does each team need to import the library?

I think the answer is yes - each team need to copy the library into their team.

**Comment by stackingsaunter** [comment]
>Is it possible to pull in a new version of a library and have all references update accordingly?

I think it's not possible - I may be wrong though, but I haven't succeeded in doing so

**Comment by GBKS** [comment]
I saw in the [export help page](https://help.penpot.app/user-guide/import-export/), that you can export multiple Penpot files as a single .penpot file. So let's say we have a main file with all the screens and 3 shared library files. When you want to download the main file, it will bundle all libraries in as well. If you then import this again, you get 4 files added to your account. That is pretty crucial for letting people easily clone this multi-file setup.

![](https://help.penpot.app/img/importing-libraries.png)
